io_context/include/msg_converters/example_interfaces.hpp

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 LeoDrive, Copyright 2021 The Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MSG_CONVERTERS__EXAMPLE_INTERFACES_HPP_
+#define MSG_CONVERTERS__EXAMPLE_INTERFACES_HPP_
+
+#include <example_interfaces/msg/u_int8_multi_array.hpp>
+
+#include "io_context/common.hpp"
+
+using example_interfaces::msg::UInt8MultiArray;
+
+namespace drivers
+{
+namespace common
+{
+
+void from_msg(const UInt8MultiArray::SharedPtr & in, MutBuffer & out);
```

Comment:
Does the API support messages that are not wrapped in a shared pointer?

Reply:
It probably makes more sense to pass in a de-referneced pointer from a `::SharedPtr` and not have to do anything for the non-`::SharedPtr` version than to have to wrap the non-`::SharedPtr` version in one. Done.